app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -134,6 +107,62 @@ function() {
         return $boltHelper->transmit('orders', $orderRequest);
     }
 
+
+    /**
+     * Get Promise to create an order token
+     *
+     * @param string  $checkoutType  'multi-page' | 'one-page' | 'admin' | 'firecheckout'
+     *
+     * @return string javascript Promise string used for initializing BoltCheckout
+     */
+    protected function getBoltOrderTokenPromise($checkoutType) {
+
+        if ( $checkoutType === self::CHECKOUT_TYPE_FIRECHECKOUT ) {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl('boltpay/order/firecheckoutcreate');
+            $validPreconditionsCheck = 'isFireCheckoutFormValid';
+            $postData = 'checkout.getFormData ? checkout.getFormData() : Form.serialize(checkout.form, true)';
+        } else if ( $checkoutType === self::CHECKOUT_TYPE_ADMIN ) {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl("adminhtml/sales_order_create/create/checkoutType/$checkoutType", true);
+            $validPreconditionsCheck = ( $checkoutType === self::CHECKOUT_TYPE_ADMIN ) ? 'isAdminFormValid' : 'true';
+            $postData = "'true'";
+        } else {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl("boltpay/order/create/checkoutType/$checkoutType");
+            $validPreconditionsCheck = 'true';
+            $postData = "'true'";
+        }
+
+        return <<<PROMISE
+                    new Promise( 
+                        function (resolve, reject) {
+                            var checkoutAjaxId = setInterval(
+                                function() {
+                                     if ($validPreconditionsCheck) {
```

Comment:
hmm, this seems hacky.. can we instead listen for form state change and call `BoltCheckout.configure` whenever needed? I assume we'll need that logic anyway to deal with change in form data?

Reply:
@daisy1754 
This is doing exactly what your are saying.  It is listen for a state change.  When the state is valid it moves on.  There are different checkout context, and so this is supplied by PHP on page render depending on the context.

I highly disagree with the classification as "hacky", and the assumption that there will always be a form is incorrect.